src/company_cmd.cpp

```diff
@@ -705,11 +707,15 @@ void OnTick_Companies()
 	}
 
 	if (_next_competitor_start == 0) {
-		_next_competitor_start = AI::GetStartNextTime() * DAY_TICKS;
+		_next_competitor_start = max(1, AI::GetStartNextTime() * DAY_TICKS);
 	}
 
-	if (AI::CanStartNew() && _game_mode != GM_MENU && --_next_competitor_start == 0) {
-		MaybeStartNewCompany();
+	if (--_next_competitor_start == 0) {
+		do {
+			if (!AI::CanStartNew() || _game_mode == GM_MENU || !MaybeStartNewCompany()) {
+				break;
+			}
+		} while (AI::GetStartNextTime() == 0);
```

Comment:
Looks like you now allow multiple AIs to possibly start in the same tick. This is probably fine but you should mention this somewhere.

Reply:
Can you move this comment to just before the do {} block?